### PR TITLE
Blenderapp v0

### PR DESCRIPTION
**Pull request by Krigpl**
note: Dockerfile final size to be optimized

### CODE REVIEW

File: tests/simulationbase.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
+from pathlib import Path
+import abc
+import json
+import random
+import shutil
+
+
+class SimulationBase:
```
Comment (Wiezzel) [inline tests/simulationbase.py]: Shouldn't it inherit from `ABC`?

File: image/entrypoints/scripts/commands/compute.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    result_dir.mkdir()
+    for rid in params['resources']:
+        with zipfile.ZipFile(network_resources_dir / f'{rid}.zip', 'r') as zipf:
+            zipf.extractall(resources_dir)
+
+    params['scene_file'] = resources_dir / params['scene_file']
+    params['crops'] = [{
+        'outfilebasename': 'result',
+        'borders_x': [params['borders'][0], params['borders'][2]],
+        'borders_y': [params['borders'][1], params['borders'][3]],
```
[the remark this answers is not in the document]
Comment (Krigpl) [reply image/entrypoints/scripts/commands/compute.py]: `compute` and `verify` require different formats of these borders, so it's either here or there. Why? Dunno, ask CGI team.

File: image/entrypoints/scripts/commands/utils.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    )
+    rows = cursor.fetchall()
+    return list(map(lambda r: r[0], rows))
+
+
+def get_next_pending_subtask(db) -> Optional[int]:
+    cursor = db.cursor()
+    cursor.execute(
+        'SELECT num FROM subtask_status WHERE status = ? LIMIT 1',
+        (SubtaskStatus.PENDING.value,)
```
[the remark this answers is not in the document]
Comment (Krigpl) [reply image/entrypoints/scripts/commands/utils.py]: I assumed we don't really care.

File: image/entrypoints/scripts/commands/verify.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        return
+    utils.set_subtask_status(db, subtask_num, utils.SubtaskStatus.FINISHED)
+    _collect_results(
+        db,
+        subtask_num,
+        task_params,
+        params,
+        subtask_results_dir,
+        results_dir,
+    )
```
[the remark this answers is not in the document]
Comment (Krigpl) [reply image/entrypoints/scripts/commands/verify.py]: App's internal consistency is still something to be more looked at. But yeah, I guess verdict should be saved nevertheless in this case.

File: tests/simulationbase.py (hunk 1 above)
Comment (Krigpl) [reply tests/simulationbase.py]: What does it change?

File: image/entrypoints/scripts/commands/compute.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Wiezzel) [reply image/entrypoints/scripts/commands/compute.py]: Understood. Then it's fine. I mean, it's not fine but let's leave like this.

File: tests/simulationbase.py (hunk 1 above)
Comment (Wiezzel) [reply tests/simulationbase.py]: [The docs](https://docs.python.org/3/library/abc.html#abc.abstractmethod) state that you should use it:
>  @abc.abstractmethod
> ...
> Using this decorator requires that the class’s metaclass is ABCMeta or is derived from it.
But I never really investigated it.

**Pull request merged**